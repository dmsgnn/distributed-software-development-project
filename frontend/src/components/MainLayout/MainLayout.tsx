import { LogoSvg } from '@/icon/Logo';
import { Box, BoxProps, Container, Flex, FlexProps, Link } from '@chakra-ui/react';
import NextLink from 'next/link';
import React, { FC } from 'react';

interface IMainLayoutProps extends FlexProps {
	navigation?: React.ReactNode;
	containerProps?: BoxProps;
}

export const MainLayout: FC<IMainLayoutProps> = ({ navigation, children, containerProps, ...rest }) => {
	return (
		<Flex direction="column" minH="100vh" {...rest}>
			<Container as="header" maxW="container.lg">
				<Flex py={6} align="center" gap={8}>
					<Link as={NextLink} href="/" aria-label="Home">
						<LogoSvg boxSize={10} />
					</Link>

					{navigation}
				</Flex>
			</Container>

			<Box flex={1} as="main" {...containerProps}>
				{children}
			</Box>
		</Flex>
	);
};
