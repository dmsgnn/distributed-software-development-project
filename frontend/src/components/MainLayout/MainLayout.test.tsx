import React from 'react';
import { axe } from 'jest-axe';

import { MainLayout } from './MainLayout';
import { act, render, screen } from 'test-utils';

describe('MainLayout', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<MainLayout />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<MainLayout />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});
		expect(results).toHaveNoViolations();
	});

	it('should render navigation', async () => {
		const navigationContent = 'navigation';
		render(<MainLayout navigation={<nav>{navigationContent}</nav>} />);

		expect(screen.getByRole('navigation')).toBeDefined();
	});

	it('should render children', async () => {
		const children = 'children';
		render(
			<MainLayout>
				<div>{children}</div>
			</MainLayout>
		);

		expect(screen.getByText(children)).toBeDefined();
	});
});
