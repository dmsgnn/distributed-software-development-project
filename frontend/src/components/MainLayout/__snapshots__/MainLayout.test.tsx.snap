// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-zf0iqh"
  >
    <header
      class="chakra-container css-f91va8"
    >
      <div
        class="css-ceb67"
      >
        <a
          aria-label="Home"
          class="chakra-link css-0"
          href="/"
        >
          <svg
            class="chakra-icon css-1iupk5b"
            focusable="false"
            viewBox="0 0 38 39"
          >
            <path
              d="M0 .5h19a19 19 0 1 1-19 19V.5Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </header>
    <main
      class="css-1rr4qq7"
    />
  </div>
</DocumentFragment>
`;
